// react-lorem-image/dist/utils.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
exports.generateURL = generateURL;
var SERVICES = {
  vanilla: function vanilla(_ref) {
    var width = _ref.width;
    var height = _ref.height;

    return 'http://www.nicenicejpg.com/' + width + '/' + height;
  },
  murray: function murray(_ref2) {
    var width = _ref2.width;
    var height = _ref2.height;
    var color = _ref2.color;

    return 'http://www.fillmurray.com/' + (color ? '' : 'g/') + width + '/' + height;
  },
  kitten: function kitten(_ref3) {
    var width = _ref3.width;
    var height = _ref3.height;
    var color = _ref3.color;

    return 'https://placekitten.com/' + (color ? '' : 'g/') + width + '/' + height;
  },
  segal: function segal(_ref4) {
    var width = _ref4.width;
    var height = _ref4.height;
    var color = _ref4.color;

    return 'http://www.stevensegallery.com/' + (color ? '' : 'g/') + width + '/' + height;
  },
  cage: function cage(_ref5) {
    var width = _ref5.width;
    var height = _ref5.height;
    var color = _ref5.color;
    var crazy = _ref5.crazy;

    var type = crazy ? 'c/' : !color ? 'g/' : '';
    return 'http://www.placecage.com/' + type + width + '/' + height;
  },
  vaughn: function vaughn(_ref6) {
    var width = _ref6.width;
    var height = _ref6.height;

    return 'http://placevaughnandfranco.it/img/' + width + '/' + height;
  },
  bear: function bear(_ref7) {
    var width = _ref7.width;
    var height = _ref7.height;
    var color = _ref7.color;

    return 'http://placebear.com/' + (color ? '' : 'g/') + width + '/' + height;
  },
  beerhold: function beerhold(_ref8) {
    var width = _ref8.width;
    var height = _ref8.height;
    var color = _ref8.color;

    return 'http://beerhold.it/' + width + '/' + height + (color ? '' : '/g');
  }
};

var TYPES = Object.keys(SERVICES);

exports.TYPES = TYPES;
var MUSTACHE = 'http://mustachify.me/?src=';

function getRandom(arr) {
  return arr[Math.floor(Math.random() * arr.length)];
}

function generateURL() {
  var _ref9 = arguments.length <= 0 || arguments[0] === undefined ? {} : arguments[0];

  var _ref9$color = _ref9.color;
  var color = _ref9$color === undefined ? true : _ref9$color;
  var _ref9$width = _ref9.width;
  var width = _ref9$width === undefined ? 200 : _ref9$width;
  var _ref9$height = _ref9.height;
  var height = _ref9$height === undefined ? width : _ref9$height;
  var _ref9$mustache = _ref9.mustache;
  var mustache = _ref9$mustache === undefined ? false : _ref9$mustache;
  var _ref9$crazy = _ref9.crazy;
  var crazy = _ref9$crazy === undefined ? false : _ref9$crazy;
  var _ref9$type = _ref9.type;
  var type = _ref9$type === undefined ? getRandom(TYPES) : _ref9$type;

  var url = SERVICES[type]({ color: color, width: width, height: height, crazy: crazy });
  return {
    type: type,
    width: width,
    height: height,
    url: mustache ? MUSTACHE + url : url
  };
}
